modules/tab_parameter_estimation/mixture_models/mixmod_regression/mixmod_regression_fun.R

```diff
@@ -0,0 +1,64 @@
+mixmod_regression_fun_ui <- function(id, model_name) {
+  ns <- shiny::NS(id)
+
+  r_function(
+    name = "rank_regression",
+    varname = model_name,
+    r_function_arg(
+      "x"
+    ),
+    r_distribution_arg(
+      inputId = ns("distribution")
+    ),
+    r_conf_level_arg(
+      inputId = ns("conf_level")
+    )
+  )
+}
+
+mixmod_regression_fun_server <- function(id, .values, estimate_cdf_r) {
+  shiny::moduleServer(
+    id,
+    function(input, output, session) {
+
+      ns <- session$ns
+
+      weibull_level <- c(0.9, 0.95, 0.99)
+      shiny::observeEvent(input$conf_level, {
+        if (input$distribution %in% c("weibull", "weibull3")) {
+          if (!input$conf_level %in% weibull_level) {
+            abs_diff <- abs(weibull_level - input$conf_level)
+            nearest <- weibull_level[abs_diff == min(abs_diff)][1]
+
+            shiny::updateNumericInput(
+              inputId = "conf_level",
+              value = nearest
+            )
+          }
+        }
+      })
+
```

Comment:
See rank_regression.R for an updated version of this code. I implemented your suggestion with a selectInput for `distribution %in% c("weibull", "weibull3")`